### PR TITLE
Add ability to add options to --static

### PR DESCRIPTION
**Pull request by albinekb**
Do you think this is a good option to add?
I think the help could be better? 🤔

### CODE REVIEW

File: bin/now-deploy.js [hunk 1]
```diff
@@ -116,7 +116,7 @@ const help = () => {
 
     --npm                     Node.js application
     --docker                  Docker container
-    --static                  Static file hosting
+    --static ${chalk.underline('OPTIONS')} Static file hosting, optional options to be passed to 'serve'
```
Comment (littleStudent) [inline bin/now-deploy.js]: maybe something more like
```
Static file hosting, options can be passed to 'serve' (e.g. --single)
```

"optional options..."  sounds strange?

Comment (albinekb) [reply bin/now-deploy.js]: 👍

**Comment by leo** [comment]
Thank you for the effort you've put into this, @albinekb! But we had a long discussion about this internally and came to the conclusion that we need stop stop extending the functionality of a the CLI and rather tell people to change their way of handling their projects to the good.

Why?

Just like mentioned [here](https://zeit.co/docs/deployment-types/static), the "static" deployment type is only meant to be used for uploading files and static websites/apps. If you're building an app that requires any special options for `serve` (and in turn a `package.json` file inside the build directory), you should not be using the "static" deployment type.

Instead, you should take advantage of a CLI (like ember-cli, create-react-app, next.js, vue-cli - something like this exists for every framework on the market) which handles the building and serving for you **on the server**.

**Comment by albinekb** [comment]
Yeah, I think that's best too :) added this because @littlestudent wanted it, so for discussion, didn't put more than a minute into this so no time wasted 🤘

**Comment by littleStudent** [comment]
@leo I see your point. I am still wondering how we should handle static SPAs?
If I have my built `dist/` folder that i want to deploy with `serve dist/ --single`. How would I achieve this without having `now` go through `npm install`?

**Comment by leo** [comment]
@littleStudent The platform should handle all of this. Don't deploy `dist/`, deploy your entire single page application project. Our goal is to be faster than localhost, and we are already (dependency caching, very fast builds). In turn, I don't see a reason for building locally.

**Comment by littleStudent** [comment]
The problem is that I have no control (except if I use Docker) over the machine. So e.g. my `elm` projects will not work because the build step will fail. So I have to build them locally / on a CI of my choosing and serve the resulting `dist/` folder directly.

**Comment by leo** [comment]
@littleStudent In that case, you'll have to add a custom `package.json` with `serve` in the "start" script, which serves your files.

**Comment by kristojorg** [comment]
@leo the downside I see here is that I would like to deploy a site as a SPA, and doing so with `serve` will run an instance of `serve`, which (1) takes up one of my 10 instances I get on the premium now plan, (2) can be frozen if there isn't enough traffic, (3) is slower than deploying over CDN. I personally would rather deploy my CRA app as a static site over CDN with a `--single` command.

**Comment by kristojorg** [comment]
@leo the downside I see here is that I would like to deploy a site as a SPA, and doing so with serve will run an instance of serve, which (1) takes up one of my 10 instances I get on the premium now plan, (2) can be frozen if there isn't enough traffic. I personally would rather deploy my CRA app as a static site with a --single command.

**Comment by leo** [comment]
In this case, you would need a an app that exports it properly (like `next export` - [more](https://github.com/zeit/next.js)). In all other cases, that's actually intentional, as we somehow need to pay our bills... 😊

**Pull request closed without merging**